apps/google-docs/functions/agents/documentParserAgent/documentParser.agent.ts

```diff
@@ -144,17 +176,64 @@ function extractTextFromGoogleDocsJson(document: unknown): string {
                   if (itemObj.paragraph && typeof itemObj.paragraph === 'object') {
                     const para = itemObj.paragraph as Record<string, unknown>;
                     if (Array.isArray(para.elements)) {
+                      let paragraphText = '';
                       for (const elem of para.elements) {
                         if (typeof elem === 'object' && elem !== null) {
                           const elemObj = elem as Record<string, unknown>;
                           if (elemObj.textRun && typeof elemObj.textRun === 'object') {
                             const textRun = elemObj.textRun as Record<string, unknown>;
                             if (typeof textRun.content === 'string') {
-                              textParts.push(textRun.content);
+                              const content = textRun.content as string;
+                              const style = (textRun.textStyle || {}) as any;
+                              const isBold = !!style.bold;
+                              const isItalic = !!style.italic;
+                              const isUnderline = !!style.underline;
+                              let wrapped = content;
+                              // Hyperlink
+                              const href = style?.link?.url as string | undefined;
+                              if (href) {
+                                const safe = String(href).replace(/"/g, '&quot;');
+                                wrapped = `<A href="${safe}">${wrapped}</A>`;
+                              }
```

Comment:
The text transformation logic is mixed with the parsing logic. Consider extracting the transformation/wrapping into separate a helper function to improve separation of concerns and make the code easier to maintain.